### ISSUE TITLE
Vulnerability in verification ok token

### ISSUE THREAD

**Issue by AbdullahSoomroOfficial** [auth0/node-jsonwebtoken#930]
**Please do not report security vulnerabilities here**. The [Responsible Disclosure Program](https://auth0.com/whitehat) details the procedure for disclosing security issues.

**Thank you in advance for helping us to improve this library!** Please read through the template below and answer all relevant questions. Your additional work here is greatly appreciated and will help us respond as quickly as possible. For general support or usage questions, use the [Auth0 Community](https://community.auth0.com/) or [Auth0 Support](https://support.auth0.com/). Finally, to avoid duplicates, please search existing Issues before submitting one here.

By submitting an Issue to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

> Provide a clear and concise description of the issue, including what you expected to happen.

### Reproduction

> Detail the steps taken to reproduce this error, what was expected, and whether this issue can be reproduced consistently or if it is intermittent.
>
> Where applicable, please include:
>
> - Code sample to reproduce the issue
> - Log files (redact/remove sensitive information)
> - Application settings (redact/remove sensitive information)
> - Screenshots

### Environment

> Please provide the following:

- **Version of this library used:**
- **Version of the platform or framework used, if applicable:**
- **Other relevant versions (language, server software, OS, browser):**
- **Other modules/plugins/libraries that might be involved:**
[jsonWebToken.txt](https://github.com/auth0/node-jsonwebtoken/files/12384455/jsonWebToken.txt)

**Comment by ArielPrevu3D** [comment]
This behavior should be better documented. This library simply uses [createPublicKey](https://nodejs.org/api/crypto.html#cryptocreatepublickeykey) which detects the key type and automatically derives the public key from private keys. 

I would not consider this to be a security issue since the key derivation only occurs when the key material is known to represent a private key.

**Comment by AbdullahSoomroOfficial** [comment]
Thank you for the guidance. @ArielPrevu3D

**Comment by guda-art** [comment]
Until now, I cannot understand the advantages of using private key signature and public key verification for the use of jwt
can  you tell me thanks !
@AbdullahSoomroOfficial

**Comment by AbdullahSoomroOfficial** [comment]
@guda-art
Public/Private key(Public key cryptography) is one of the way we can use to do two things:
1. To encrypt data
2. To create digital signature

For example In a web app if we use jwt for authentication.
In that case we do not want to encrypt any data.
But instead we want to sign a token(To create digital signature) to ensure data integrity and authenticity.
For that we first hash the data and sign it with our private key and issue it to client.
Later on the client side we can verify the token using our public key.
From verification we will get two surities:
1. Issuer of the token is the one who it says it is.
2. Data is tempered or not.

So by using public/private key we can securely transmit any data over http

**Comment by guda-art** [comment]
Thank you very much. I think I understand because my project does not require client-side verification of server information, so I feel confused. This way, I understand
@AbdullahSoomroOfficial

**Comment by AbdullahSoomroOfficial** [comment]
@guda-art 
What is your project about. May be I could help and learn more from you.

**Comment by guda-art** [comment]
@AbdullahSoomroOfficial 
Yes, the requirement for my project is to send the signed information from the server to the client, and then the server will perform a verification when it is brought back by the client. There is no need for the client to participate in the verification, so I only need a key, which is symmetric encryption. thanks.
